chef/cookbooks/provisioner/recipes/update_nodes.rb

```diff
@@ -377,6 +377,16 @@ def find_node_boot_mac_addresses(node, admin_data_net)
         cloud_available = true if name.include? "Cloud"
       end
 
+      cpu_model = ""
+      if mnode[:cpu]
```

Comment:
Shouldn't you check here, if `mnode[:cpu]["0"]` and `mnode[:cpu]["0"][:model_name]` exist?